File: elixir-elasticemail-examples/examples/suppressions.exs
# Usage: mix run examples/suppressions.exs [email]
email = List.first(System.argv()) || "suppressed@example.com"
email_path = ElasticEmail.path_segment(email)

fail = fn step, error ->
  IO.puts(:stderr, "Error (#{step}): #{ElasticEmail.format_error(error)}")
  System.halt(1)
end

# Suppressions are split into unsubscribes, bounces and complaints.
# Adding to any list stops future sends to that address. The body is a JSON array.
case ElasticEmail.post("/suppressions/unsubscribes", [email]) do
  {:ok, _} -> IO.puts("Added to unsubscribes: #{email}")
  error -> fail.("add unsubscribe", error)
end

case ElasticEmail.get("/suppressions/#{email_path}") do
  {:ok, s} ->
    IO.puts(
      "Suppression: " <>
        inspect(%{
          Email: s["Email"],
          Reason: s["FriendlyErrorMessage"],
          DateUpdated: s["DateUpdated"]
        })
    )

  error ->
    fail.("get suppression", error)
end

case ElasticEmail.get("/suppressions", limit: 10, offset: 0) do
  {:ok, suppressions} ->
    IO.puts("\nAll suppressions (first #{length(suppressions)}):")
    Enum.each(suppressions, fn s -> IO.puts(" - #{s["Email"]} #{s["FriendlyErrorMessage"] || ""}") end)

  error ->
    fail.("list suppressions", error)
end

# Remove it again so the address can receive email
case ElasticEmail.delete("/suppressions/#{email_path}") do
  {:ok, _} -> IO.puts("\nRemoved from suppressions: #{email}")
  error -> fail.("delete suppression", error)
end

File: dotnet-elasticemail-examples/Examples/Suppressions.cs
namespace ElasticEmailExamples;

using ElasticEmail.Api;
using ElasticEmail.Client;

public static class Suppressions
{
    // Usage: dotnet run -- suppressions [email]
    public static async Task RunAsync(string[] args)
    {
        var suppressionsApi = new SuppressionsApi(Ee.Config());
        var email = args.Length > 0 ? args[0] : "suppressed@example.com";

        // Suppressions are split into unsubscribes, bounces and complaints.
        // Adding to any list stops future sends to that address.
        try
        {
            await suppressionsApi.SuppressionsUnsubscribesPostAsync(new List<string> { email });
            Console.WriteLine($"Added to unsubscribes: {email}");
        }
        catch (ApiException e)
        {
            Ee.Fail("add unsubscribe", e);
        }

        try
        {
            var s = await suppressionsApi.SuppressionsByEmailGetAsync(email);
            Console.WriteLine($"Suppression: Email={s.Email} Reason={s.FriendlyErrorMessage} DateUpdated={s.DateUpdated:O}");
        }
        catch (ApiException e)
        {
            Ee.Fail("get suppression", e);
        }

        try
        {
            var all = await suppressionsApi.SuppressionsGetAsync(10, 0);
            Console.WriteLine($"\nAll suppressions (first {all.Count}):");
            foreach (var s in all)
            {
                Console.WriteLine($" - {s.Email} {s.FriendlyErrorMessage}");
            }
        }
        catch (ApiException e)
        {
            Ee.Fail("list suppressions", e);
        }

        // Remove it again so the address can receive email
        try
        {
            await suppressionsApi.SuppressionsByEmailDeleteAsync(email);
            Console.WriteLine($"\nRemoved from suppressions: {email}");
        }
        catch (ApiException e)
        {
            Ee.Fail("delete suppression", e);
        }
    }
}
